### DISCUSSION TITLE
External router settings

### DISCUSSION THREAD

**Discussion by WimDeMeester** [WimDeMeester/eFinder#56] [Q&A] [answered]
Hi @AstroKeith,

I received my external router.  What settings do I need?  I can connect to the router on the Nexus DSC Pro.  I take a  fixed address there.  The IP address is in the 192.168.0 range.  Do I need to change something in the code?  I tried to enter the fixed IP address there, but I can not connect from the Raspberry Pi to the Nexus.  Do you have an idea what I'm doing wrong?

**Comment by AstroKeith** [comment]
I’m away just now and can’t view code with line numbers.  In the __init__ function in Nexus.py you will find the IP that the Nexus is expected to be found on. 10.0.0.1. Change this to whatever your router has assigned.

**Reply by WimDeMeester** [reply, marked as answer]
Changing 10.0.0.1 in Nexus.py indeed fixed the problem.  However, I also had to make sure the Nexus DSC Pro had the correct settings.  This is what I entered (and now it works):

- WiFi Mode: Infrastructure
- SSID: the name of the D-link router network
- Password: Select the password
- Channel: 6
- IP address: 192.168.0.19
- Mask: 255.255.255.0
- Gateway: 192.168.0.1
- Protocol: LX200 (ServoCAT will not connect with the eFinder)

The router is set up to only use channel 6 and to hand out DHCP addresses in the 192.168.0.20 - 192.168.0.60 range.

**Comment by AstroKeith** [comment]
You might find that the router changes the IP it assigns to the Nexus if powered down. I’ve started to disable DHCP on the Nexus and set a static IP.

BTW in infrastructure mode, the Nexus ignores the channel number as it takes it from the server.

**Reply by WimDeMeester** [reply]
Thanks!  I use a static IP address for the Nexus, so that should be fine.  The Raspberry PI and my tablet are getting a DHCP address, but that doesn't matter - I guess...

**Comment by WimDeMeester** [comment]
Hi @AstroKeith , 
I made a few easy changes to be able to set the IP address of the Nexus in the eFinder.config file.  Are you interested in including these changes in your code?  In that case, I will make sure you receive the changes.
